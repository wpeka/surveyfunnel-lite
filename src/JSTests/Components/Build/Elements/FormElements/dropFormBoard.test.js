import React, { useCallback, useRef, useState } from "react";
import { HTML5Backend } from 'react-dnd-html5-backend'
import { DndProvider } from 'react-dnd'

import DropFormBoard from '../../../../../Components/Build/Elements/FormElements/DropFormBoard';

import renderer from 'react-test-renderer';

const ele = {
    id: "simple",
    text: "simple",
    index: "dfd",
    item: "sdf",
    moveCard: "dfd",
    editList: (item) => { test1 = item },
    deleteFromList: (item) => { test1 = "" },
    itemType: 'simple',
    componentName: 'Sample',
    name: 'sample',
}
const ele2 = {
    id: "2simple",
    text: "simple",
    index: "dfd",
    item: "sdf",
    moveCard: "dfd",
    editList: (item) => { test1 = item },
    deleteFromList: (item) => { test1 = "" },
    itemType: 'simple',
    componentName: 'Sample',
    name: 'sample',
}
const list=[

];
test('DropFormBoard elements test', async () => {

    const onchange = () => {
        return;
    }
    const useDrag = jest.fn();

    window.React = React
    const component = renderer.create(<DndProvider backend={HTML5Backend}> <DropFormBoard ele={ele} List={ele.list} editList={ele.editList} deleteFromList={ele.deleteFromList} moveCard={ele.moveCard}/> </DndProvider>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});