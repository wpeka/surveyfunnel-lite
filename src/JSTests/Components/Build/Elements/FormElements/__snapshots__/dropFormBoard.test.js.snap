// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropFormBoard elements test 1`] = `
Array [
  " ",
  <div
    className="surveyfunnel-lite-dropboard"
    role="DropBoard"
    style={
      Object {
        "backgroundColor": "#F4EAFC",
      }
    }
  >
    Drag a box here
    <h2>
      sample
    </h2>
    <div
      className="showBoard"
    >
      <div
        className="cardBox"
        style={
          Object {
            "cursor": "move",
            "opacity": 1,
          }
        }
      >
        <div
          className="surveyfunnel-lite-cardbox-title"
        >
          <img
            src="test-file-stub"
          />
          <h3>
            sample
          </h3>
        </div>
        <div
          className="card-flex"
        >
          <button
            className="surveyfunnel-lite-cardBox-btn"
            onClick={[Function]}
          >
            <img
              src="test-file-stub"
            />
          </button>
          <button
            className="surveyfunnel-lite-cardBox-btn"
            onClick={[Function]}
          >
            <img
              src="test-file-stub"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;
